linera-base/src/crypto/mod.rs

```diff
@@ -50,14 +49,19 @@ pub enum CryptoError {
     )]
     IncorrectHashSize(usize),
     #[error(
-        "Byte slice has length {0} but a `PublicKey` requires exactly {expected} bytes",
-        expected = dalek::PUBLIC_KEY_LENGTH,
+        "Byte slice has length {len} but a {scheme} `PublicKey` requires exactly {expected} bytes"
     )]
-    IncorrectPublicKeySize(usize),
+    IncorrectPublicKeySize {
+        scheme: String,
+        len: usize,
+        expected: usize,
+    },
     #[error("Could not parse integer: {0}")]
     ParseIntError(#[from] ParseIntError),
     #[error("secp256k1 error: {0}")]
-    Secp256k1Error(::secp256k1::Error),
+    Secp256k1Error(k256::ecdsa::Error),
+    #[error("could not parse public key: {0}: key point at inifinity.")]
+    Secp256k1PointAtIfinity(String),
```

Comment:
```suggestion
    #[error("could not parse public key `{0}`: point at infinity")]
    Secp256k1PointAtInfinity(String),
```
(re-commenting suggestion since it was partially applied)

Reply:
Done in [Apply review suggestions.](https://github.com/linera-io/linera-protocol/pull/3407/commits/8f530c26831f602c5a89abc74abc7445a02389d6).

Reply:
The modified version you committed still has trailing punctuation, against the `Error` guidelines :sweat_smile: 

The reason for this guideline is that it's common to embed the error message inside a larger error string at a caller site up the stack, and having trailing punctuation breaks the grammar, which can cause it to be difficult to decide where one error ends and the next begins when reading the error.  This hurts user (or developer) experience.

Reply:
OK but let's not introduce new guidelines for one variant out of 10. If we want to agree how it should be done let's all agree on some common approach and introduce it for everything in a single PR. If I do what you're suggesting then this is still inconsistent and who's making a decision which one is better.